extensions/scheduler/common/src/main/java/io/quarkus/scheduler/common/runtime/DefaultInvoker.java

```diff
@@ -14,12 +14,8 @@ public CompletionStage<Void> invoke(ScheduledExecution execution) throws Excepti
         if (requestContext.isActive()) {
             return invokeBean(execution);
         } else {
-            try {
-                requestContext.activate();
-                return invokeBean(execution);
-            } finally {
-                requestContext.terminate();
-            }
+            requestContext.activate();
+            return invokeBean(execution).whenComplete((v, t) -> requestContext.terminate());
```

Comment:
Is it possible for `invokeBean` to throw an exception before returning a `CompletionStage`? If that's the case, you still need a `try-catch`  to make sure that `requestContext.terminate()` is called.

Reply:
You're right!